packages/utils/src/lib/report.ts

```diff
@@ -56,3 +63,26 @@ export function compareIssueSeverity(
 export function sumRefs(refs: CategoryRef[]) {
   return refs.reduce((sum, { weight }) => sum + weight, 0);
 }
+
+export function loadReports(options: PersistConfig): [string, string][] {
+  const { outputDir, filename, format } = options;
+  ensureDirectoryExists(outputDir);
+
+  return (
+    readdirSync(outputDir)
+      // filter by file extension
+      .filter(file =>
+        format ? format.find(ext => file.endsWith(`.${ext}`)) : true,
+      )
+      // filter by file name
+      .filter(file =>
+        filename
+          ? file.includes(filename)
+          : new RegExp(REPORT_NAME_PATTERN).test(file),
+      )
+      .map(file => {
+        const filePath = join(outputDir, file);
+        return [file, readFileSync(filePath, 'utf8')];
+      })
+  );
+}
```

Comment:
Again, async `fs` APIs. This will read the files one-by-one, can easily run in parallel with `Promise.all`.